misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt

```diff
@@ -0,0 +1,52 @@
+package misk.redis
+
+import misk.time.FakeClock
+import java.time.Instant
+import javax.inject.Inject
+
+class FakeRedis : Redis {
+  @Inject lateinit var clock: FakeClock
+
+  // The value type stored in our key-value store
+  private data class Value (
+    val value: String,
+    val expiryInstant: Instant = Instant.MAX
+  )
+
+  // Acts as the Redis key-value store
+  private val keyValueStore = HashMap<String, Value>()
```

Comment:
you probably want this to be thread safe, since a test might use multiple threaads

Reply:
a simple `ConcurrentHashMap` is fine to use

Reply:
also FYI for the future. you typically want to use `mutableMap`, instead of `HashMap` directly. It's more idiomatic and It will use a `LinkedHashMap` under the hood, which is more test friendly. 

https://publicobject.com/2016/02/08/linkedhashmap-is-always-better-than-hashmap/

Reply:
Good call, using `ConcurrentHashMap`